packages/iov-bcp-types/src/connection.ts

```diff
@@ -280,9 +280,10 @@ export interface BcpConnection {
   /**
    * Looks up transaction in history and if not found, waits until it is available.
    *
-   * As a consequence, this never resolves for non-existing transaction IDs.
+   * As a consequence, this sends an event for non-existing transaction IDs. Stream completes
+   * after 1 element was sent.
    */
-  readonly waitForTransaction?: (id: TransactionId) => Promise<ConfirmedTransaction | FailedTransaction>;
+  readonly waitForTransaction?: (id: TransactionId) => Stream<ConfirmedTransaction | FailedTransaction>;
```

Comment:
What's up with this change?
Seems to make it harder to use, and I don't see the upside.

Reply:
Have a look at the 2 resolves conversation in here. The issue is you cannot cancel a Promise.

Reply:
> toListPromise(..., 1) and waiting for stream completion will also block in the same manner.

Right, every promise can be converted to a stream without losing information. However, not every stream can be converted into a promise. What `toListPromise` does is removing the power to unsibscribe, which is okay for tests that will timeout.